packages/gatsby-plugin-gatsby-cloud/src/gatsby-node.js

```diff
@@ -30,14 +31,24 @@ exports.onCreateWebpackConfig = ({ actions, stage }) => {
   })
 }
 
-exports.onPostBuild = async (
-  { store, pathPrefix, reporter },
-  userPluginOptions
-) => {
+exports.onPostBuild = async ({ store, pathPrefix }, userPluginOptions) => {
   const pluginData = makePluginData(store, assetsManifest, pathPrefix)
+
   const pluginOptions = { ...DEFAULT_OPTIONS, ...userPluginOptions }
 
-  const { redirects, pageDataStats, nodes } = store.getState()
+  const { redirects, pageDataStats, nodes, pages } = store.getState()
+
+  /**
+   * Emit via IPC routes for which pages are non SSG
+   */
+  for (const [pathname, page] of pages) {
+    if (page.mode && page.mode !== `SSG`) {
+      emitRoutes({
+        [`${pathname.substring(1)}index.html`]: page.mode,
+        [`page-data/${pathname.substring(1)}page-data.json`]: page.mode,
```

Comment:
In practice this won't work because you we use https://github.com/gatsbyjs/gatsby/blob/087338898edf5bce88149aad702325969ce3742e/packages/gatsby/src/utils/page-data.ts#L36-L43 to construct the path for page-data.json and for HTML files we use https://github.com/gatsbyjs/gatsby/blob/087338898edf5bce88149aad702325969ce3742e/packages/gatsby/src/utils/page-html.ts#L8-L19 to construct the path.

So e.g. if you add a test case for `/` path your test should fail at the moment.

Because `/` will be `index`:

https://github.com/gatsbyjs/gatsby/blob/087338898edf5bce88149aad702325969ce3742e/packages/gatsby/src/utils/page-data.ts#L28-L30